core/src/main/java/hudson/security/HudsonPrivateSecurityRealm.java

```diff
@@ -358,7 +358,7 @@ private User createAccount(StaplerRequest req, StaplerResponse rsp, boolean self
         if(si.fullname==null || si.fullname.length()==0)
             si.fullname = si.username;
 
-        if(si.email==null || !si.email.contains("@"))
+        if(isMailerPluginPresent() && (si.email==null || !si.email.contains("@")))
```

Comment:
Agree, but given the timing and existing code, I think handling of this is probably sufficient: this at least hides the fields that would not save properly, which would be very confusing and buggy to the user